### ISSUE TITLE
Hotfix bug in BIDS2Legacy for large datasets

### ISSUE THREAD

**Issue by jan-petr** [ExploreASL/ExploreASL#1276]
### Release v1.10.1

Hotfix for BIDS2Legacy of large datasets. 
Release issue to document testing and bug fixing. Check out the GitHub wiki workflow for the [drafting of a new release](https://github.com/ExploreASL/ExploreASL/wiki/Drafting-a-new-Release).

### Hotfix error

- [x] Avoid loading BIDS structure for every subject, but do it only once.
- [x] Avoid printing report for all subjects for each subject, but print only those done in a single iteration.
- [x] Test if processing a large dataset is fast enough.

### Testing

Only start testing once the release branch is created. Test all again after all major fixes.

#### Run UnitTesting - (xASL_test_Unittesting) - all should be passed

- [x] Linux: Jan

#### Run TestDatasets - (xASL_test_TestExploreASL) - post the final table here in this issue

- [x] Linux: Jan

#### Run Flavors test - (xASL_test_Flavors([],0,0) - Run only the import part including BIDS2Legacy and stop before processing. Paste the printed errors.

- [x] Linux: Jan

### Administration

Do these tasks in the release branch while testing:

- [x] Document `CHANGES.md` using release notes of all issues: **n/a**
- [x] Update `VERSION` file: **n/a**

### Release

Start the release process only once all the tasks above are done:
- [x] Merge release branch to main: **n/a**
- [x] Rebase develop with latest commits: **n/a**
- [x] Update version to BETA in develop: **n/a**
- [x] Do the release, copy release text, tag, and Zenodo: **n/a**

### Release notes
Fix error in BIDS2Legacy conversion execution time.

**Comment by MauricePasternak** [comment]
This probably needs to be re-opened. The old, slow behavior still occurs if someone wants to re-run certain subjects with the Import Module using `ExploreASL('some/path', 1, 0)`. Which is odd...since this shouldn't run in that case???

Basically, imagine trying to run the Import Module for one extra subject, but before it is reached you need to wait through many minutes of:

```
DCM2NIIX was run before...   
NIIX to ASL-BIDS was run before... 78%
DCM2NIIX was run before...   
NIIX to ASL-BIDS was run before... 79%
DCM2NIIX was run before...   
NIIX to ASL-BIDS was run before... 79%
DCM2NIIX was run before...   
NIIX to ASL-BIDS was run before... 80%
DCM2NIIX was run before.
```

**Comment by jan-petr** [comment]
I have created a new issue for this #1281
